client/src/App.jsx

```diff
@@ -0,0 +1,88 @@
+const React = require("react");
+const ReactDOM = require("react-dom");
+const $ = require("jquery");
+
+import Gallery from "./components/Gallery.jsx";
+import ProductInfo from "./components/ProductInfo.jsx";
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+
+    this.getData = function(prodNum, appObj) {
+      prodNum = prodNum || 1;
+
+      var reqProduct = $.get({
+        url: "http://localhost:9001/get",
+        context: this,
+        data: { id: prodNum }
+      });
+      reqProduct.done(res => {
+        this.setState(res[0]);
```

Comment:
is this the right format for setState? typically Ive seen setState({key: value})